### PR TITLE
fix: no-explicit-any rule

### PR DESCRIPTION
**Pull request by tuliomir**
This PR is composed of two separate commits for an easier scoped review: one commit for the nano-contracts code and the other for the remaining lib code.

### Acceptance Criteria
- Fixes [`@typescript-eslint/no-explicit-any`](https://typescript-eslint.io/rules/no-explicit-any/) rule

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by r4mmer — APPROVED** [review]

File: src/nano_contracts/types.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export enum NanoContractActionType {
   WITHDRAWAL = 'withdrawal',
 }
 
+export type NanoContractArgumentType = string | Buffer | number | boolean;
```
Comment (pedroferreira1) [inline src/nano_contracts/types.ts:14]: This can be `null` as well.

File: src/nano_contracts/types.ts [hunk 2]
```diff
@@ -73,7 +75,7 @@ export interface NanoContractHistoryAPIResponse {
 
 interface StateValueSuccess {
   // State value return
-  value: any;
+  value: NanoContractArgumentType;
```
Comment (pedroferreira1) [inline src/nano_contracts/types.ts:78]: This can't be `Buffer` because it comes from the API

File: src/nano_contracts/builder.ts [hunk 3]
```diff
@@ -34,7 +39,7 @@ class NanoContractTransactionBuilder {
 
   caller: Buffer | null;
 
-  args: any[] | null;
+  args: NanoContractArgumentType[] | null;
```
Comment (pedroferreira1) [inline src/nano_contracts/builder.ts:42]: We don't expect `Buffer` here

File: src/nano_contracts/serializer.ts [hunk 4]
```diff
@@ -160,7 +161,7 @@ class Serializer {
    * @memberof Serializer
    * @inner
    */
-  fromOptional(value: any, type: string): Buffer {
+  fromOptional(value: NanoContractArgumentType, type: string): Buffer {
```
Comment (pedroferreira1) [inline src/nano_contracts/serializer.ts:164]: This `value` can be `null` as well

File: src/nano_contracts/types.ts (hunk 1 above)
Comment (tuliomir) [reply src/nano_contracts/types.ts]: Added on https://github.com/HathorNetwork/hathor-wallet-lib/pull/685/commits/ae048f821e498c422ba6450dd422a10f0cd7198c

File: src/nano_contracts/types.ts (hunk 2 above)
Comment (tuliomir) [reply src/nano_contracts/types.ts]: On https://github.com/HathorNetwork/hathor-wallet-lib/pull/685/commits/ae048f821e498c422ba6450dd422a10f0cd7198c I added another type named `NanoContractArgumentApiInputType` that has no Buffer. I believe this solves this issue

File: src/nano_contracts/builder.ts (hunk 3 above)
Comment (tuliomir) [reply src/nano_contracts/builder.ts]: Changed to the new type mentioned above.

File: src/nano_contracts/serializer.ts (hunk 4 above)
Comment (tuliomir) [reply src/nano_contracts/serializer.ts:164]: This was solved by editing the type directly, as said above.

**Review by pedroferreira1 — APPROVED** [review]

**Review by r4mmer — APPROVED** [review]

**Pull request merged**